src/sessions/RevocationModal.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useState } from "react";
+import {
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  Form,
+  FormGroup,
+  Modal,
+  ModalVariant,
+  TextContent,
+  TextInput,
+  ValidatedOptions,
+} from "@patternfly/react-core";
+import { useTranslation } from "react-i18next";
+import { useForm } from "react-hook-form";
+
+import type UserRepresentation from "keycloak-admin/lib/defs/userRepresentation";
+import { emailRegexPattern } from "../util";
+import type RealmRepresentation from "keycloak-admin/lib/defs/realmRepresentation";
+import { useAdminClient, useFetch } from "../context/auth/AdminClient";
+import moment from "moment";
+import { useRealm } from "../context/realm-context/RealmContext";
+import type ClientRepresentation from "keycloak-admin/lib/defs/clientRepresentation";
+import { useAlerts } from "../components/alert/Alerts";
+
+type RevocationModalProps = {
+  id?: string;
+  handleModalToggle: () => void;
+  activeClients: ClientRepresentation[];
+  save: (realm?: UserRepresentation) => void;
+};
+
+export const RevocationModal = ({
+  handleModalToggle,
+  save,
+  activeClients,
+}: RevocationModalProps) => {
+  const { t } = useTranslation("sessions");
+  const { addAlert } = useAlerts();
+
+  const { realm: realmName } = useRealm();
+  const adminClient = useAdminClient();
+  const { register, errors, handleSubmit } = useForm();
+  const [realm, setRealm] = useState<RealmRepresentation>();
+
+  const [key, setKey] = useState(0);
+
+  const refresh = () => {
+    setKey(new Date().getTime());
+  };
+
+  useFetch(
+    async () => {
+      const realm = await adminClient.realms.findOne({ realm: realmName });
+
+      return { realm };
+    },
+    ({ realm }) => {
+      setRealm(realm);
+    },
```

Comment:
Can be written a bit simpler:

```suggestion
  useFetch(
    () => adminClient.realms.findOne({ realm: realmName }),
    (realm) => {
      setRealm(realm);
    },
```